src/OpenTK.Core/Platform/Interfaces/IClipboardComponent.cs

```diff
@@ -4,13 +4,65 @@
 using System.Text;
 using System.Threading.Tasks;
 
+#nullable enable
+
 namespace OpenTK.Core.Platform
 {
     // GOAL: We want to support unicode text, rich text and images.
     public interface IClipboardComponent : IPalComponent
     {
+        /// <summary>
+        /// A list of formats that this clipboard component supports.
+        /// </summary>
+        IReadOnlyList<ClipboardFormat> SupportedFormats { get; }
+
+        /// <summary>
+        /// Gets the format of the data currently in the clipboard.
+        /// </summary>
+        /// <returns>The format of the data currently in the clipboard.</returns>
+        ClipboardFormat GetClipboardFormat();
+
+        /// <summary>
+        /// Sets the string currently in the clipboard.
+        /// </summary>
+        /// <param name="text">The text to put on the clipboard.</param>
         void SetClipboardText(string text);
 
-        string GetClipboardText();
+        /// <summary>
+        /// Returns the string currently in the clipboard.
+        /// This function returns null if the current clipboard data doesn't have the <see cref="ClipboardFormat.Text"/> format.
+        /// </summary>
+        /// <returns>The string currently in the clipboard, or null.</returns>
+        string? GetClipboardText();
+
+        /// <summary>
+        /// Gets the audio data currently in the clipboard.
+        /// This function returns null if the current clipboard data doesn't have the <see cref="ClipboardFormat.Audio"/> format.
+        /// </summary>
+        /// <returns>The audio data currently in the clipboard.</returns>
+        // FIXME: Figure out the return type!
+        object? GetClipboardAudio();
+
+        /// <summary>
+        /// Gets the bitmap currently in the clipboard.
+        /// This function returns null if the current clipboard data doesn't have the <see cref="ClipboardFormat.Bitmap"/> format.
+        /// </summary>
+        /// <returns>The bitmap currently in the clipboard.</returns>
+        // FIXME: Figure out the return type!
+        object? GetClipboardBitmap();
```

Comment:
If and when we create a Bitmap object, we should revisit IIconComponent and modify it to accommodate the changes. I would prefer that we convert the images to a raw bitmap (much like how OpenGL processes them without compressed formats).

Reply:
I have an initial API for this. We likely want to make it more complete, but its something for now.

It also needs documentation.